internal/awslambda/extension/telemetry/server.go

```diff
@@ -0,0 +1,97 @@
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/sirupsen/logrus"
+)
+
+type RuntimeDoneHook func()
+
+func NoopHook() RuntimeDoneHook {
+	return func() {}
+}
+
+type Server struct {
+	addr string
+	log  logrus.FieldLogger
+	f    RuntimeDoneHook
```

Comment:
`context.CancelFunc` is reasonable to do here.

Reply:
I planned on having the whole lifecycle controlled via `Run(context Context)` , what's the additional value of storing the cancelFunc?